### DISCUSSION TITLE
Latest vedo version with `snake_case` naming scheme

### DISCUSSION THREAD

**Discussion by marcomusy** [marcomusy/vedo#705] [General]
Hi `vedo` users :)
as the python community is slowly but steadily abandoning the `camelCase` convention for variable names in favor of the most widely adopted `snake_case` convention I have updated methods  and keywords names in the library to follow the PEP8 guidelines.
Hence code like this (in vedo==2022.3.1):
```python
mesh = Mesh("bunny.obj")
mesh.cutWithPlane()
```
now becomes (in vedo==2022.4.1):
```python
mesh = Mesh("bunny.obj")
mesh.cut_with_plane()     ##### <----
```
To make the transition as painless as possible I tried to add as many warnings as possible and to maintain backward compatibility, but there might be cases when your code is no more backward compatible (the changes needed are only a renaming, by substituting uppercase names with an underscore _ and lowercase letter).

Let me know if you encounter any difficulty in the process!
Thanks for your patience!

@gioda @jo-mueller @RubendeBruin @manu @ivan-marroquin  @ManuGraiph @antmatyjajo

**Comment by jo-mueller** [comment]
Hi @marcomusy ,

thanks for tagging me here, I'll be on the lookout 👍

**Comment by ZiguoAtGitHub** [comment]
I saw one mistake in renaming just now. The descriptions of return values of fit function in the api docs on web had changed but the libs doesn't. For example, fitd.data_sigma should be kept fitd.dataSigma in coding.
Just a reminder.🙂

**Comment by baba-yaga** [comment]
Hopefully vtk won't change to snake_case convention any time soon - I have already had much fun with adapting old vedo codes 🤪
